applications/wasm-opt-for-rust.md

```diff
@@ -0,0 +1,299 @@
+though they probably do.
+These bindings appear to rely on CMake to build.
+We do not expect to use them directly, but may reference them during development.
+
+[brs]: https://github.com/pepyakin/binaryen-rs
+
+The [`cargo-wasi`] tool takes a [different strategy][cws] to acquiring `wasm-opt`.
+It automatically downloads the `wasm-opt` binary, presumably from the official releases.
+Other wasm toolchains like `cargo-contract` could follow a similar strategy,
+possibly by extracting the existing logic from `cargo-wasi`.
+
+[`cargo-wasi`]: https://github.com/bytecodealliance/cargo-wasi
+[cws]: https://bytecodealliance.github.io/cargo-wasi/wasm-opt.html#which-wasm-opt-executed
+
+
+
+
+
+## Team :busts_in_silhouette:
+
+### Team members
+
+- **Team lead:** [Brian Anderson](https://github.com/brson)
+- **Team member:** [Aimee Zhu](https://github.com/aimeedeer)
+
+### Contact
+
+- **Contact Name:** Brian Anderson
+- **Contact Email:** andersrb@gmail.com
+- **Website:** https://brson.github.io
+
+### Legal Structure
+
+- **Registered Address:** 16192 Coastal Highway, Lewes, Delaware 19958
+- **Registered Legal Entity:** Common Orbit LLC
+
+### Team's experience
+
+The team lead is one of the original authors of the Rust programming language,
+with 12 years of Rust experience.
+They have performed Rust work for Mozilla, Reddit, PingCAP, Solana, MobileCoin, Parity, and Nervos.
+Both team members are maintainers of the Rust in Blockchain newsletter.
+
+### Team Code Repos
+
+- https://github.com/brson/wasm-opt-rs
+- https://github.com/brson
+- https://github.com/aimeedeer
+
+### Team LinkedIn Profiles (if available)
+
+N/A
+
+
+
+
+## Development Status :open_book:
+
+https://github.com/brson/wasm-opt-rs
+
+We have created the initial project layout and investigated the feasibility of building the binaryen codebase using only the [`cc`] crate,
+as well as the feasibility of trivially calling the `wasm-opt` `main` function from Rust as described.
+
+[`cc`]: https://docs.rs/cc/latest/cc/
+
+We have reserved the `wasm-opt` and `wasm-opt-sys` crate names on crates.io.
+
+
+
+
+## Development Roadmap :nut_and_bolt:
+
+### Overview
+
+- **Total Estimated Duration:** 3-4 months
+- **Full-Time Equivalent (FTE):**  0.4
+- **Total Costs:** 30,000 USD
+
+This will be a part time effort.
+I have divided this into two logical milestones,
+though in reality the work will overlap.
+
+
+### Milestone 1 — Proof of Concept
+
+- **Estimated duration:** 1-2 month
+- **FTE:**  0.4
+- **Costs:** 15,000 USD
+
+During this phase we will prove the concept and produce a `wasm-opt` Rust binary and API.
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | MIT / Apache-2.0 |
+| 0b. | Documentation | Basic README. |
+| 0c. | Testing Guide | Manual smoke testing on Linux, Windows, and MacOS, x86_64 and ARM. |
+| 0d. | Docker | Docker is not required for this project. |
+| 0e. | Article | None. |
```

Comment:
```suggestion
```

If you're not planning to deliver an article on M1, we can just skip this delivery.